### DISCUSSION TITLE
Option to set a default BIOS for emulated systems.

### DISCUSSION THREAD

**Discussion by TsukiZero** [mamedev/discussions#126] [Ideas]
I know I can just like create a `neogeo.ini` and add `bios unibios40`, but it would be nice if in the MAME GUI there were some sort of  "emulated system settings" where I could set the system's BIOS to UniBIOS 4.0 instead of having to manually make the aforementioned INI, or having to manually go through each game of that emulated system.

**Comment by sairuk** [comment]
technically already there
1. select system/machine
2. select bios in system settings, save system settings (`mslug2.ini`)
3. turn on skip bios menu and save the general settings (`ui.ini`)

however you'll need to reconfigure your rom paths for the system ini tho, they are not carried across to the system ini file which I presume is a bug in the ini file reading somewhere

**Comment by cuavas** [comment]
It wouldn’t really make sense given BIOS options are different for every system.

**Reply by sairuk** [reply]
with that in mind, and probably already known ... my thoughts after have a bit of a look last night

`skip_biosmenu` and some other ui opts should probably become system options even though they are ui elements. 

`skip_biosmenu` is currently stored in `ui.ini` which from what i can tell it doesn't have a system equivalent nor perhaps should it ... arrived at my conclusion as `ui.ini` seems to be hardcoded in a few places and therefore setting it for one system in `ui.ini` affects bios selection for all. 

I expect there is probably a better longer term vision than just allowing folks to "make a mslug2_ui.ini" for example, so i'm happy enough to have acknowledged the behaviour here, wont speak for others though.

**Reply by cuavas** [reply]
The reason that setting is in ui.ini is partly because at the time it shows the BIOS selection menu, it hasn’t set up the selected machine and loaded its INI files (it’s still running the `__empty` machine under the menu at that point).  The current options system code really doesn’t lend itself to changing that, for much the same reasons that cause a bunch of other quirks/issues.

**Reply by sairuk** [reply]
I was digging around earlier today and did note we are still under `__empty` at this point in time. It makes sense at startup of the ui to be here and the options I've referred to don't make much sense to be part of a dedicated system ini as a result. Honestly `__empty` was not something i'd given much mind too in the past.

e.g. for other folks who may come across this, running mame with verbose (or check title bar in ui) refers to the `__empty` machine
```
$ ./mame -w -v
Attempting load of mame.ini
Parsing mame.ini
Attempting load of mame.ini
Starting plugin layout...
Starting plugin data...
Starting plugin hiscore...
Attempting load of mame.ini
Parsing mame.ini
Attempting load of mame.ini
Attempting load of source/empty.ini
Attempting load of ___empty.ini
```

I understand these are just quirks of the current implementation and unfortunately I don't have a good answer for a way forward at the moment, however if we get a ui refactor in the future it is probably worth considering how we implement these kind of options for usability.
